File: Python3/LayerCake.py
import Cake

class LayerCake(Cake):
    def __init__(self, flavor, frosting, numLayers):
        super().__init__((9.00 + ((numLayers-1) * 3) + (numLayers if frosting.lower() == "cream-cheese" else 0)), flavor, frosting)
        self.numLayers = numLayers

    def __str__(self):
        return self.numLayers + "-layer " + super().getFlavor() + " cake with " + super().getFrosting() + " frosting " + super

    #Overriding abstract DiscountedPrice(quantity) from BakedGood
    def DiscountedPrice(self, quantity):
        return ((super().getBasePrice() - 2.00) if quantity >= 3 else (super().getBasePrice())) * quantity

File: Dart/bakery_checkout/bin/layer_cake.dart
import 'cake.dart';

class LayerCake extends Cake {
  final int _numLayers;

  LayerCake(String flavor, String frosting, this._numLayers)
      : super(
            (9.00 +
                ((_numLayers - 1) * 3) +
                (frosting == "cream-cheese" ? _numLayers : 0)),
            flavor,
            frosting);

  @override
  String toString() {
    return "$_numLayers-layer ${super.flavor} cake with ${super.frosting} frosting " +
        super.toString();
  }

  @override
  double discountedPrice(int quantity) {
    return (quantity >= 3 ? super.basePrice - 2.00 : super.basePrice) *
        quantity;
  }
}
